bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/discovery/LiquidCheckDiscoveryService.java

```diff
@@ -0,0 +1,217 @@
+        } catch (UnknownHostException e) {
+            logger.debug("UnknownHostException {}", e.toString());
+        } catch (IOException e) {
+            logger.debug("IOException {}", e.toString());
+        } catch (Exception e) {
+            logger.debug("Exception {}", e.toString());
+        }
+    }
+
+    /**
+     * Method to stop the scan
+     */
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    /**
+     * Method for creating a Runnable to start a scan
+     * 
+     * @return the Runnable
+     */
+    protected Runnable liquidCheckDiscoveryRunnable() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                startScan();
+            }
+        };
+        return runnable;
+    }
+
+    /**
+     * This Method retrieves all IPv4 addresses of the server
+     * 
+     * @return A list of all available IPv4 Adresses that are registered
+     * @throws SocketException
+     */
+    private List<InetAddress> getIPv4Adresses() throws SocketException {
+        Iterator<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces().asIterator();
+        List<InetAddress> addresses = new ArrayList<>();
+        // Get IPv4 addresses from all network interfaces
+        if (null != networkInterfaces) {
+            while (networkInterfaces.hasNext()) {
+                NetworkInterface currentNetworkInterface = networkInterfaces.next();
+                Iterator<InetAddress> inetAddresses = currentNetworkInterface.getInetAddresses().asIterator();
+                while (inetAddresses.hasNext()) {
+                    InetAddress currentAddress = inetAddresses.next();
+                    if (currentAddress instanceof Inet4Address && !currentAddress.isLoopbackAddress()) {
+                        addresses.add(currentAddress);
+                    }
+                }
+            }
+        }
+        return addresses;
+    }
+
+    /**
+     * This method will find any active host in the network and return a list of them
+     * 
+     * @param addresses
+     * @return List of hosts
+     * @throws UnknownHostException
+     * @throws IOException
+     */
+    private List<InetAddress> findActiveHosts(List<InetAddress> addresses) throws UnknownHostException, IOException {
+        List<InetAddress> hosts = new ArrayList<>();
+        for (InetAddress inetAddress : addresses) {
+            String[] adresStrings = inetAddress.getHostAddress().split("[.]");
+            String subnet = adresStrings[0] + "." + adresStrings[1] + "." + adresStrings[2];
+            int timeout = 50;
+            for (int i = 1; i < 255; i++) {
+                String host = subnet + "." + i;
+                if (InetAddress.getByName(host).isReachable(timeout)) {
+                    hosts.add(InetAddress.getByName(host));
```

Comment:
Maybe you can immediately exclude the OH server, this will avoid a useless HTTP request in `startScan`.